frontend/src/scenes/surveys/SurveySQLHelper.tsx

```diff
@@ -0,0 +1,153 @@
+import { useValues } from 'kea'
+import { router } from 'kea-router'
+import { CodeSnippet, Language } from 'lib/components/CodeSnippet'
+import { IconOpenInNew } from 'lib/lemon-ui/icons'
+import { surveyLogic } from 'scenes/surveys/surveyLogic'
+import { urls } from 'scenes/urls'
+
+import { NodeKind } from '~/queries/schema/schema-general'
+import { Survey, SurveyQuestion } from '~/types'
+
+import { createAnswerFilterHogQLExpression, getResponseFieldCondition } from './utils'
+
+interface SurveySQLHelperProps {
+    isOpen: boolean
+    onClose: () => void
+}
+
+export function SurveySQLHelper({ isOpen, onClose }: SurveySQLHelperProps): JSX.Element {
+    const { survey, answerFilters } = useValues(surveyLogic)
+
+    const filterConditions = createAnswerFilterHogQLExpression(answerFilters, survey as Survey)
+
+    const generateSingleQuestionQuery = (question: SurveyQuestion, index: number): string => {
+        return `SELECT
+    distinct_id,
+    ${getResponseFieldCondition(index, question.id)} AS "${question.question}",
+    timestamp
+FROM
+    events
+WHERE
+    event = 'survey sent'
+    AND properties.$survey_id = '${survey.id}'${filterConditions ? '\n' + filterConditions : ''}
+ORDER BY
+    timestamp DESC
+LIMIT
+    100`
+    }
+
+    const generateFullSurveyQuery = (): string => {
+        const questionSelects = survey.questions
+            .map((question: SurveyQuestion, index: number) => {
+                return `    ${getResponseFieldCondition(index, question.id)} AS "${question.question}"`
+            })
+            .join(',\n')
+
+        return `SELECT
+    distinct_id,
+${questionSelects},
+    timestamp
+FROM
+    events
+WHERE
+    event = 'survey sent'
+    AND properties.$survey_id = '${survey.id}'${filterConditions ? '\n' + filterConditions : ''}
+ORDER BY
+    timestamp DESC
+LIMIT
+    100`
+    }
+
+    // Function to open query in a new insight
+    const openInInsight = (query: string): void => {
+        const insightQuery = {
+            kind: NodeKind.DataTableNode,
+            full: true,
+            source: {
+                kind: NodeKind.HogQLQuery,
+                query: query,
+            },
+        }
+        router.actions.push(urls.insightNew({ query: insightQuery }))
+    }
+
+    // Check if any filters are applied
+    const hasAppliedFilters = (): boolean => {
+        return answerFilters && answerFilters.some((f) => f.value && (!Array.isArray(f.value) || f.value.length > 0))
+    }
+
+    return (
+        <LemonModal
+            isOpen={isOpen}
+            onClose={onClose}
+            title="SQL Query Helper"
+            description={
+                <div className="flex flex-col gap-1 text-sm text-muted">
+                    <p>
+                        <b>Important:</b> Since March 7, 2024, survey responses are stored using question IDs instead of
+                        indices. The queries below handle both formats using the <code>coalesce</code> function.
```

Comment:
```suggestion
                        indexes. The queries below handle both formats using the <code>coalesce</code> function.
```

not 100% sure but I think indices is more common in statistics only